src/services/OrderService.ts

```diff
@@ -0,0 +1,339 @@
+        viewIndex++;
+      } else {
+        throw resp.data;
+      }
+    }
+    while (resp.data.docs.length >= 250);
+  } catch (error) {
+    logger.error(error);
+  }
+  return orderItems
+}
+
+const fetchOrderItemStats = async (orderItemsList: any): Promise<any> => {
+  const orderItems = orderItemsList;
+  const shippedQtyRequests = [], receivedQtyRequests = [];
+  const orderitemStats = {} as any;
+
+  while(orderItems.length) {
+    const batch = orderItems.splice(0,100);
+    const orderIds = [] as any, orderItemSeqIds = [] as any, productIds = [] as any;
+
+    batch.map((itemMap: any) => {
+      const itemInfo = itemMap.split("_");
+      orderIds.push(itemInfo[0])
+      orderItemSeqIds.push(itemInfo[1])
+      productIds.push(itemInfo[2])
+    })
+
+    const shippedQtyParams = {
+      inputFields: {
+        orderId: orderIds,
+        orderId_op: "in",
+        orderItemSeqId: orderItemSeqIds,
+        orderItemSeqId_op: "in"
+      },
+      viewSize: 250,
+      entityName: 'ShippedItemQuantitySum',
+      fieldList: ['orderId', 'orderItemSeqId', 'shippedQuantity']
+    }
+
+    const receivedQtyParams = {
+      inputFields: {
+        orderId: orderIds,
+        orderId_op: "in",
+        orderItemSeqId: orderItemSeqIds,
+        orderItemSeqId_op: "in"
+      },
+      viewSize: 250,
+      entityName: 'ReceivedItemQuantitySum',
+      fieldList: ['orderId', 'orderItemSeqId', 'totalQuantityAccepted']
+    }
+    shippedQtyRequests.push(shippedQtyParams)
+    receivedQtyRequests.push(receivedQtyParams)
+  }
+
+  const shippedItemQtyResps = await Promise.allSettled(shippedQtyRequests.map((params) => api({
+    url: 'performFind',
+    method: 'POST',
+    data: params
+  })))
+
+  const receivedItemQtyResps = await Promise.allSettled(receivedQtyRequests.map((params) => api({
+    url: 'performFind',
+    method: 'POST',
+    data: params
+  })))
+
+  shippedItemQtyResps.map((response: any) => {
+    if(response.status === "fulfilled" && !hasError(response.value) && response.value.data?.docs?.length) {
+      response.value.data.docs.map((doc: any) => {
+        orderitemStats[`${doc.orderId}_${doc.orderItemSeqId}`] = { shippedQty: doc.shippedQuantity }
+      })
+    }
+  })
+
+  receivedItemQtyResps.map((response: any) => {
+    if(response.status === "fulfilled" && !hasError(response.value) && response.value.data?.docs?.length) {
+      response.value.data.docs.map((doc: any) => {
+        if(orderitemStats[`${doc.orderId}_${doc.orderItemSeqId}`]) {
+          orderitemStats[`${doc.orderId}_${doc.orderItemSeqId}`] = { ...orderitemStats[`${doc.orderId}_${doc.orderItemSeqId}`], receivedQty: doc.totalQuantityAccepted }
+        } else {
+          orderitemStats[`${doc.orderId}_${doc.orderItemSeqId}`] = { receivedQty: doc.totalQuantityAccepted }
+        }
+      })
+    }
+  })
+
+  return orderitemStats
+}
+
+const fetchFacilityAddresses = async (facilityIds: any): Promise<any> => {
```

Comment:
We can store the facility in state and persist the same, as this is not something that will change much frequently, so if we have a facility info in state then instead of making api call, we can return the data from state.